### PR TITLE
Enable multi-platform development

### PR DESCRIPTION
**Pull request by nickgerace**
TLDR: remove barriers to multi-platform development, like Bash and Make.
Moreover, remove the nightly fmt dependency and improve testing to catch
potential corner cases on multiple platforms.

Core:
- Rename "Reports" to "LabeledReports" since "Reports" is not of implied
  type Vec<Report>

Testing:
- Add second half to the integration test: comparing found reports with
  expected reports
  - Essentially, we run the core gfold loop _again_, but only generate
    reports and skip displaying them to stdout

CI:
- Removed nightly fmt check and job
- Moved fmt check to stable
- Update bors toml to match changes

Scripts:
- Convert scripts from Bash scripts to independent crates
  (multi-platform friendly)
- Add README to scripts directory
- Ensure script crates do not have their Cargo lockfile tracked in Git

Docs:
- Adding DEVELOPING file to replace Makefile
- Move THANKS and RELEASE files to new docs directory, along with
  DEVELOPING
- Replace make commands in README and RELEASE files

Misc:
- Remove Rust nightly dependency since it was solely for imports
  granulaity for rustfmt
  - This repository is once again only reliant on stable Rust
    toolchains
- Remove "rust-version" from Cargo toml since it was unused
- Remove Makefile entirely in favor of DEVELOPING file (multi-platform
  friendly)
- Update markdown code block languages as "shell" instead of "bash" to
  be multi-platform friendly

### CODE REVIEW

**Comment by nickgerace** [comment]
bors merge

File: src/report/mod.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ pub struct Report {
 }
 
 impl Report {
-    fn new(
+    pub fn new(
```
Comment (nickgerace) [inline src/report/mod.rs:44]: Made public for the integration test

File: src/report/mod.rs [hunk 2]
```diff
@@ -19,10 +19,10 @@ const HEAD: &str = "HEAD";
 /// parent directory for a group of reports ([`Vec<Report>`]). The values corresponding to those keys
 /// are the actual groups of reports.
 // NOTE: We use a BTreeMap over a HashMap for sorted keys.
-pub type Reports = BTreeMap<Option<String>, Vec<Report>>;
+pub type LabeledReports = BTreeMap<Option<String>, Vec<Report>>;
 
 /// A collection of results for a Git repository at a given path.
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
```
Comment (nickgerace) [inline src/report/mod.rs:25]: Required for `pretty_assertions::assert_eq!`

File: src/status.rs [hunk 3]
```diff
@@ -3,7 +3,7 @@
 use serde::{Deserialize, Serialize};
 
 /// A summarized interpretation of the status of a Git working tree.
-#[derive(Debug, Clone, Copy, Deserialize, Serialize)]
+#[derive(Debug, Clone, Copy, Deserialize, Serialize, PartialEq)]
```
Comment (nickgerace) [inline src/status.rs:6]: Required for `pretty_assertions::assert_eq!`
